OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -249,11 +258,43 @@ internal void UnreserveSpace(Actor a)
 
 			reservedWeight -= GetWeight(a);
 			reserves.Remove(a);
+			ReleaseLock(self);
 
 			if (loadingToken != ConditionManager.InvalidConditionToken)
 				loadingToken = conditionManager.RevokeCondition(self, loadingToken);
 		}
 
+		// Prepare for transport pickup
+		bool LockForPickup(Actor self)
+		{
+			if (state == State.Locked)
+				return false;
+
+			state = State.Locked;
+
+			self.CancelActivity();
```

Comment:
Do we really need to cancel the activity queue here?

The current/previous code paused the transport while loading, but allowed it to continue moving and doing any other queued activities afterwards.

Reply:
The `Land` and `WaitFor` activities need to be inserted at the front of the queue interrupting any current activity if possible. I don't know how we can continue the previous queue where it left of without running into trouble with activity states. I'm not sure if `ChildActivities` are any help here.

Reply:
The current/old approach for locking the actor while loading was to use `LoadingCondition` to pause `Mobile`. I had always assumed that we would hook that up to `Aircraft`'s `ForceLandCondition` - this was one of the motivations for adding that in the first place.

Reply:
`ForceLandCondition` as currently implemented also cancels the activity queue so that solves nothing. I would prefer to leave this issue for a future follow-up.

Reply:
Leaving this to a followup is reasonable, but I do think the `ForceLandCondition` is the way to go - theres no conceptual need for that to be cancelling activities even though it is doing that now.